flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/table/AsyncMLPredictITCase.java

```diff
@@ -0,0 +1,302 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.stream.table;
+
+import org.apache.flink.table.api.config.ExecutionConfigOptions;
+import org.apache.flink.table.planner.factories.TestValuesModelFactory;
+import org.apache.flink.table.planner.factories.TestValuesTableFactory;
+import org.apache.flink.table.planner.runtime.utils.StreamingWithStateTestBase;
+import org.apache.flink.testutils.junit.extensions.parameterized.ParameterizedTestExtension;
+import org.apache.flink.testutils.junit.extensions.parameterized.Parameters;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CollectionUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.planner.factories.TestValuesTableFactory.changelogRow;
+import static org.assertj.core.api.Assertions.assertThatList;
+
+/** ITCase for async ML_PREDICT. */
+@ExtendWith(ParameterizedTestExtension.class)
+public class AsyncMLPredictITCase extends StreamingWithStateTestBase {
+
+    private final Boolean objectReuse;
+    private final ExecutionConfigOptions.AsyncOutputMode asyncOutputMode;
+
+    public AsyncMLPredictITCase(
+            StateBackendMode backend,
+            Boolean objectReuse,
+            ExecutionConfigOptions.AsyncOutputMode asyncOutputMode) {
+        super(backend);
+
+        this.objectReuse = objectReuse;
+        this.asyncOutputMode = asyncOutputMode;
+    }
+
+    private final List<Row> data =
+            Arrays.asList(
+                    Row.of(1L, 12, "Julian"),
+                    Row.of(2L, 15, "Hello"),
+                    Row.of(3L, 15, "Fabian"),
+                    Row.of(8L, 11, "Hello world"),
+                    Row.of(9L, 12, "Hello world!"));
+
+    private final List<Row> dataWithNull =
+            Arrays.asList(
+                    Row.of(15L, null, "Hello"),
+                    Row.of(3L, 15, "Fabian"),
+                    Row.of(11L, null, "Hello world"),
+                    Row.of(9L, 12, "Hello world!"));
+
+    private final List<Row> cdcRowData =
```

Comment:
Should CDC be supported if ml_predict output is non-deterministic?

Reply:
I think it's a good feature for some users. If some users cares about the non-deterministic, we can support this later.